airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/metadata.py

```diff
@@ -209,9 +209,34 @@ def _run(self, connector: Connector) -> CheckResult:
         return self.pass_(connector=connector, message="The upgrade deadline is set to at least a week in the future")
 
 
+class CheckConnectorMaxSecondsBetweenMessagesValue(MetadataCheck):
+    name = "Certified source connector must have a value filled out for maxSecondsBetweenMessages in metadata"
+    description = "Certified source connectors must have a value filled out for `maxSecondsBetweenMessages` in metadata. This value represents the maximum number of seconds we could expect between messages for API connectors. And it's used by platform to tune connectors heartbeat timeout. The value must be set in the 'data' field in connector's `metadata.yaml` file."
+    applies_to_connector_types = ["source"]
+
+    @staticmethod
+    def check_connector_certified(connector: Connector) -> bool:
```

Comment:
I think this method should move up to the `Check` abstract class and you should add a `runs_on_certified_connectors` property and change `Check.run` method to skip there if the connector is not certified. (same logic as `if connector.language not in self.applies_to_connector_languages:`)